runtime/server/observability_utils.go

```diff
@@ -0,0 +1,38 @@
+package server
+
+import (
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func safeTimeStr(t *timestamppb.Timestamp) string {
+	if t == nil {
+		return ""
+	}
+	return t.AsTime().String()
+}
+
+func marshalProtoSlice[K proto.Message](s []K) []string {
+	res := make([]string, len(s))
+	for i := 0; i < len(res); i++ {
+		res[i] = marshalProto(s[i])
+	}
+	return res
+}
+
+func marshalProto(pb proto.Message) string {
+	b, err := protojson.Marshal(pb)
+	if err != nil {
+		return ""
+	}
+	return string(b)
+}
```

Comment:
It doesn't feel so clean to have encoded json fields as log fields. I'm wondering if we should either:
- Flatten messages to multiple fields with dot-separated keys, like `"parent.child.key"`
- Or just have a custom func for each pb type (there are not many). E.g. `InlineMeasures` could just return a `[]string` for `StringSlice` of the names (the other info is less important).

Reply:
I am not sure how we can flatten messages in a generic way. I can add a combination of both in the sense that when it is not possible to just return `[]string`. I can add multiple fields with dot-separated keys.